### ISSUE TITLE
Footnotes appear on the last html page instead of one they are placed on.

### ISSUE THREAD

**Issue by MateuszKalamarz** [rstudio/bookdown#793]
While creating my own gitbook as site (using RStudio's *Knit to gitbook*) I realized that footnotes end up on the very last HTML page of the whole site. I tried to use `split_bib: yes` explicitly (as far as I know it is default behaviour in HTML documents) but it didn't work. Thinking it has to do with my code and the book itself I tried using example provided for oxforddown: https://github.com/ulyngs/oxforddown. For example https://ulyngs.github.io/oxforddown/rmd-basics.html#footnotes uses footnotes and they appear on the same page. When I use *Knit to gitbook* in RStudio all footnotes end up in the end in the *References* section (which doesn't appear in the oxforddown version hosted online).

Thus it is either problem with my system or with bookdown. Hope someone can reproduce it.

Checked with newest R, RStudio and updated packages. 

Output of `xfun::session_info('bookdown')` below:

```
R version 3.6.1 (2019-07-05)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 18362), RStudio 1.2.5001

Locale:
  LC_COLLATE=Polish_Poland.1250  LC_CTYPE=Polish_Poland.1250    LC_MONETARY=Polish_Poland.1250
  LC_NUMERIC=C                   LC_TIME=Polish_Poland.1250    

Package version:
  base64enc_0.1.3 bookdown_0.14.1 digest_0.6.22   evaluate_0.14   glue_1.3.1      graphics_3.6.1  grDevices_3.6.1
  highr_0.8       htmltools_0.4.0 jsonlite_1.6    knitr_1.25      magrittr_1.5    markdown_1.1    methods_3.6.1  
  mime_0.7        Rcpp_1.0.2      rlang_0.4.1     rmarkdown_1.16  stats_3.6.1     stringi_1.4.3   stringr_1.4.0  
  tinytex_0.17    tools_3.6.1     utils_3.6.1     xfun_0.10       yaml_2.2.0     
```
---

By filing an issue to this repo, I promise that

- [x] I have fully read the issue guide at https://yihui.name/issue/.
- [x] I have provided the necessary information about my issue.
    - If I'm asking a question, I have already asked it on Stack Overflow or RStudio Community, waited for at least 24 hours, and included a link to my question there.
    - If I'm filing a bug report, I have included a minimal, self-contained, and reproducible example, and have also included `xfun::session_info('bookdown')`. I have upgraded all my packages to their latest versions (e.g., R, RStudio, and R packages), and also tried the development version: `remotes::install_github('rstudio/bookdown')`.
    - If I have posted the same issue elsewhere, I have also mentioned it in this issue.
- [x] I have learned the Github Markdown syntax, and formatted my issue correctly.

I understand that my issue may be closed if I don't fulfill my promises.

**Comment by cderv** [comment]
> While creating my own gitbook as site (using RStudio's Knit to gitbook)

I am not sure that the Knit to gitbook is the way to go. I can't reproduce your issue but I can say how it is advice to build bookdown books. 

See https://bookdown.org/yihui/bookdown/usage.html
You can use from the code `bookdown::render_book` or the Build book button in the Build pane in RStudio IDE. 

Did you already use those ? 
Do you still have issues if you build your book using these ways ? 

Thanks.

**Comment by MateuszKalamarz** [comment]
> > While creating my own gitbook as site (using RStudio's Knit to gitbook)
> 
> I am not sure that the Knit to gitbook is the way to go. I can't reproduce your issue but I can say how it is advice to build bookdown books.
> 
> See https://bookdown.org/yihui/bookdown/usage.html
> You can use from the code `bookdown::render_book` or the Build book button in the Build pane in RStudio IDE.
> 
> Did you already use those ?
> Do you still have issues if you build your book using these ways ?
> 
> Thanks.

Yes I tried the Build pane where I select only select to build bookdown::gitbook (the button says Build Book). I haven't tried `bookdown::render_book` yet.

**Comment by cderv** [comment]
> Yes I tried the Build pane where I select only select to build bookdown::gitbook

I tried with the [bookdown demo](https://github.com/rstudio/bookdown-demo) by adding a footnote in the littérature chapter and it works well. Footnote is in the same page. 
Can you try that to ? 

Or Can you submit a reproductible example of your issue ?

**Comment by MateuszKalamarz** [comment]
> > Yes I tried the Build pane where I select only select to build bookdown::gitbook
> 
> I tried with the [bookdown demo](https://github.com/rstudio/bookdown-demo) by adding a footnote in the littérature chapter and it works well. Footnote is in the same page.
> Can you try that to ?
> 
> Or Can you submit a reproductible example of your issue ?

I did as you mentioned.

Code of **02-literature.Rmd**:

```
# Literature

Here is a review of existing methods.

First footnote ^[testing inline footnote text.].

Second foothnote[^bib-test].

[^bib-test]: testing separately stating footnote text.
```

Both end up under *References*.

![footnote_in_references](https://user-images.githubusercontent.com/57257538/68043416-8c160e00-fcd5-11e9-8cb5-7a05881fea63.PNG)

**Comment by cderv** [comment]
The reference placed at the end of the book is an old behavior that has ben modified since 0.1
(see https://github.com/rstudio/bookdown/issues/84). You seem to have 0.14 now so it should not be related. I can't reproduce your issue currently so I can't say what is going on. That is very strange

**Comment by MateuszKalamarz** [comment]
> The reference placed at the end of the book is an old behavior that has ben modified since 0.1
> (see #84). You seem to have 0.14 now so it should not be related. I can't reproduce your issue currently so I can't say what is going on. That is very strange

I saw it, do you remember what was the issue the other time? I updated everything that I could think of. Is there some soft, package or application that could need updating/reinstalling? Also is there a way to obtain the whole rendering log-file to go line by line and spot whatever is causing this behaviour?

**Comment by cderv** [comment]
Can you confirm you ran the bookdown demo repo as is ? adding footnotes in litteratures Chapter ? 

If you want to debug interactively in your environment, (I can't do it myself as I can't reproduce)

* The footnote moving happens in `split_chapters `
* Footnotes should be parsed and removed 
https://github.com/rstudio/bookdown/blob/8d16e2a943fde2659ccbee886ebb40d226ad0ec6/R/html.R#L311-L314
* then moved to each chapter 
https://github.com/rstudio/bookdown/blob/8d16e2a943fde2659ccbee886ebb40d226ad0ec6/R/html.R#L378
* `split_chapter` is run in `gitbook`
https://github.com/rstudio/bookdown/blob/8d16e2a943fde2659ccbee886ebb40d226ad0ec6/R/gitbook.R#L55
 
You can run using `debug` or `debugonce` to stop inside one of this function and goes step by step for there. 

Hopes it helps precise the diagnosis.
If you can provide me a reproducible example, I'll help. Otherwise I let you dig into it.

**Comment by andrewgribben** [comment]
I'm also having the same issue in my book. I edited a chapter from the the bookdown demo as advised and the same thing happens there too. Both books are being built using the Build button in RStudio.

```
R version 3.6.1 (2019-07-05)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Catalina 10.15.1

Locale: en_GB.UTF-8 / en_GB.UTF-8 / en_GB.UTF-8 / C / en_GB.UTF-8 / en_GB.UTF-8

Package version:
  base64enc_0.1.3  bookdown_0.14.1  digest_0.6.21    evaluate_0.14
  glue_1.3.1       graphics_3.6.1   grDevices_3.6.1  highr_0.8
  htmltools_0.4.0  jsonlite_1.6     knitr_1.25       magrittr_1.5
  markdown_1.1     methods_3.6.1    mime_0.7         Rcpp_1.0.2
  rlang_0.4.0      rmarkdown_1.16.2 stats_3.6.1      stringi_1.4.3
  stringr_1.4.0    tinytex_0.16     tools_3.6.1      utils_3.6.1
  xfun_0.10        yaml_2.2.0
```

**Comment by cderv** [comment]
Interesting ! Can you tell me both what is your pandoc version ? 
`rmardown::pandoc _version()` ? 
There may be something off with how footnotes are handled but it seems to dépend of something on your respective environment as I can’t reproduce.

**Comment by MateuszKalamarz** [comment]
> Interesting ! Can you tell me both what is your pandoc version ?
> `rmardown::pandoc_version()` ?
> There may be something off with how footnotes are handled but it seems to dépend of something on your respective environment as I can’t reproduce.

Right now 2.7.3, was 2.7.2 at the time I posted this issue, but I did reinstall everything (R, RStudio, pandoc, even Rtools, tinytex). Nothing changed.

**Comment by andrewgribben** [comment]
> Interesting ! Can you tell me both what is your pandoc version ?
> `rmardown::pandoc_version()` ?
> There may be something off with how footnotes are handled but it seems to dépend of something on your respective environment as I can’t reproduce.

2.7.3 here

**Comment by cderv** [comment]
Thanks ! Now I can reproduce with new pandoc 2.7.3. I'll can investigate properly now. I should have begin with this question.
It seems to work fine with pandoc 2.7.2 (which I had before upgrading) - you can downgrade to make it work until a fix is pushed.

**Comment by mkalamarz** [comment]
> Thanks ! Now I can reproduce with new pandoc 2.7.3. I'll can investigate properly now. I should have begin with this question.
> It seems to work fine with pandoc 2.7.2 (which I had before upgrading) - you can downgrade to make it work until a fix is pushed.

Yes, it works with 2.7.2, thank you!

I think it is worth mentioning one thing. I had pandoc 2.7.3 installed, but in Rstudio/pandoc folder there was 2.7.2. When I checked with `rmardown::pandoc_version()` it gave me `2.7.2`. As I mentioned this was at the time I posted this issue. It looks like rmarkdown used 2.7.3 as it was found on the system. Still it can be misleading for people in my situation.

**Comment by MateuszKalamarz** [comment]
> Thanks ! Now I can reproduce with new pandoc 2.7.3. I'll can investigate properly now. I should have begin with this question.
> It seems to work fine with pandoc 2.7.2 (which I had before upgrading) - you can downgrade to make it work until a fix is pushed.

Yes, it works with 2.7.2, thank you!

I think it is worth mentioning one thing. I had pandoc 2.7.3 installed, but in Rstudio/pandoc folder there was 2.7.2. When I checked with rmardown::pandoc_version() it gave me 2.7.2. As I mentioned this was at the time I posted this issue. It looks like rmarkdown used 2.7.3 as it was found on the system. Still it can be misleading for people in my situation.

**Comment by cderv** [comment]
After a small investigation, I found the bug was caused by this change in pandoc https://github.com/jgm/pandoc/issues/5469
that was introduced in pandoc 2.7.3. 

Bookdown uses regex a lot and this change of arrow character in footnotes from one Unicode char to two Unicode char makes the process of identifying moving the footnotes breaks. 

I’ll push a quick fix later today. It will let me think if we can improve the workflow to other than regex, or at least how best improve the regex.
